.github/workflows/build_centos.yaml

```diff
@@ -0,0 +1,59 @@
+# Copyright (C) 2023 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+
+# This workflow uses actions that are not certified by GitHub.
+# This workflow builds and pushes a Docker container to GitHub 
+# Packages when a release is create or a pull request is merged to main.
+
+
+name: Create and publish a CentOS7 image
+
+on:
+  push:
+    branches:
+    - 'main'
+    - 'release-*'
+  pull_request:
+    branches:
+      - 'main'
+      - 'release-*'
```

Comment:
I think we don't want this on the release branches for push or pull request, otherwise it will roll back versions when we do point releases to stable.